week3/Program.cs

```diff
@@ -0,0 +1,132 @@
+should be a product of Rate and amount multiplication.
+Bonus: We should also check that Rate or amount are not negative.
+ */
+app.MapGet("/exchangeRate", () =>
+{
+    var amount = 150;
+    var exchangeRate = new ExchangeRate("EUR", "DKK");
+    exchangeRate.Rate = 7.5m;
+    Console.WriteLine($"The rate of {amount} {exchangeRate.FromCurrency} is {exchangeRate.Calculate(amount)} {exchangeRate.ToCurrency}");
+});
+
+/* Task 3 :
+Create Account class that can be initialized with the amount value. Account class contains Withdraw and Deposit 
+methods and Balance (get only) property. Make sure that you can't withdraw more than you have in the balance.
+ */
+
+app.MapGet("/accountBalance", () =>
+{
+    var amountBalance = new Account(500);
+    amountBalance.Deposit(500);
+    Console.WriteLine($"Account balance is {amountBalance.Balance}");
+    //amountBalance.Withdraw(300);
+    amountBalance.Withdraw(1100);
+    System.Console.WriteLine($"Remain Account balance is {amountBalance.Balance}");
+    
+});
+
+/* Task 4
+ Create interface IAnimal with property Name and Sound . Create classes Cow, Cat and Dog that implement IAnimal . 
+Instantiate all three classes and pass them to a new method called MakeSound that has single parameter IAnimal and 
+it writes to console eg “Dog says woof woof” if instance of the Dog class is passed.
+*/
+
+
+app.Run();
+
+
+//Temperature
+public class Temperature
+{
+    public decimal tempCelcius = -273.15m;
+    public decimal Celcius { get; set; }
+
+    public Temperature(decimal tempCelciusDegree)
+    {
+        Celcius = tempCelciusDegree;
+        if (Celcius < tempCelcius)
+        {
+            System.Console.WriteLine("Temperature cannot be less than -273.15");
+            throw new System.Exception("Temperature cannot be less than -273.15");
+        }
+    }
+    public string Fahrenheit
+    {
+        get => ($"{((Celcius * 9) / 5) + 32}");
+    }
+}
+
+// ExchangeRate Task
+public class ExchangeRate
+{
+    public string FromCurrency { get; set; }
+    public string ToCurrency { get; set; }
+    public decimal Rate { get; set; }
+    public ExchangeRate(string fromCurrency, string toCurrency)
+    {
+        FromCurrency = fromCurrency;
+        ToCurrency = toCurrency;
+    }
+    public decimal Calculate(decimal amount)
+    {
+        if (Rate <= 0 || amount <= 0)
+        {
+            throw new Exception("Amount shouldn't be negative");
+        }
+        var result = Rate * amount;
+        return result;
+    }
+}
+
+//Bank account
+public class Account
+{
+    public decimal Amount { get; set; }
+    public decimal Balance { get; set; }
+    public Account(decimal balance)
+    {
+        Balance = balance;
+    }
+    public decimal Deposit(decimal Amount)
+    {
+        {
+            if (Balance > 0)
+            {
+                Balance += Amount;
+            }
+            return Balance;
```

Comment:
the return value is not used so you can remove this and change the return type to `void` (Same for the withdraw method)